### PR TITLE
Gui Optimizations

### PR DESCRIPTION
**Pull request by thomasslee97**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

1. Limits tracker re-runs for segments not affected by current/next part changing.
2. Fixes issue of segments occasionally not rewinding properly.

* **What is the current behavior?** (You can also link to an open issue here)

1. Because of changes in subscribed collections, the tracker in `SegmentTimelineContainer` are running the autorun for every visible segment all the time, including running a heavy `getResolvedSegment` function.
2. Sometimes after resetting the rundown, segments get rewound but jump back to their old positions after a while.

* **What is the new behavior (if this is a feature change)?**

1. Subscriptions are limited so that only the current segment and segments changing their current/next state have their tracker re-run.
2. Segments are not jumping to the old position after rewinding.

* **Other information**:

**Status**

- [ ] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

**Comment by thomasslee97** [comment]
Needs input from @jstarpl @ianshade

File: meteor/client/ui/RundownView/RundownTiming.tsx [hunk 1]
```diff
@@ -251,7 +251,7 @@ export const RundownTimingProvider = withTracker<
 			}
 			if (prevProps.parts !== this.props.parts) {
 				// empty the temporary Part Instances cache
-				this.temporaryPartInstances = {}
+				this.onRefreshTimer()
```
Comment (jstarpl) [inline meteor/client/ui/RundownView/RundownTiming.tsx:254]: ```this.temporaryPartInstances = {}```
Should not be removed. The temporaryPartInstances is another speed-up/improvement that has been implemented at NRK and should be kept.

**Review by jstarpl — CHANGES_REQUESTED** [review]
I think there are some clashes with some other improvements done over at NRK. These optimizations should coexist with each other, so all that's necessary is to make sure to keep the original ones where they are.

File: meteor/client/ui/RundownView/RundownTiming.tsx (hunk 1 above)
Comment (jstarpl) [reply meteor/client/ui/RundownView/RundownTiming.tsx:255]: Both lines are now included.

File: meteor/client/lib/viewPort.ts [hunk 2]
```diff
@@ -90,7 +90,7 @@ export async function scrollToPart(partId: PartId, forceScroll?: boolean, noAnim
 
 const FALLBACK_HEADER_HEIGHT = 65
 let HEADER_HEIGHT: number | undefined = undefined
-export const HEADER_MARGIN = 25
+export const HEADER_MARGIN = 15 // NRK uses: 25
```
[the remark this answers is not in the document]
Comment (jstarpl) [reply meteor/client/lib/viewPort.ts]: I restored the 25 setting.

**Review by jstarpl — APPROVED** [review]
Look good

**Pull request merged**